subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright (c) 2021 Demant
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <string.h>
+#include <zephyr/types.h>
+#include <sys/types.h>
+#include <toolchain.h>
+
+#include "isoal.h"
+
+/* TBD Leave empty for now */
+#define LL_ASSERT(cond)
+
+#define MIN(a, b)         (((a) < (b)) ? (a) : (b))
+
+/* TBD this must be taken from a Kconfig */
+#define ISOAL_SINKS_MAX   (4)
+
+/** Allocation state */
+typedef uint8_t isoal_alloc_state_t;
+#define ISOAL_ALLOC_STATE_FREE            ((isoal_alloc_state_t) 0x00)
+#define ISOAL_ALLOC_STATE_TAKEN           ((isoal_alloc_state_t) 0x01)
+
+static struct
+{
+	isoal_alloc_state_t  sink_allocated[ISOAL_SINKS_MAX];
+	struct isoal_sink    sink_state[ISOAL_SINKS_MAX];
+} isoal_global;
+
+
+/**
+ * @brief Internal reset
+ * Zero-init entire ISO state
+ */
+static isoal_status_t _init_reset(void)
+{
+	memset(&isoal_global, 0, sizeof(isoal_global));
+	return ISOAL_STATUS_OK;
+}
+
+/**
+ * @brief  Initialize ISO-AL
+ */
+isoal_status_t isoal_init(void)
+{
+	isoal_status_t err = ISOAL_STATUS_OK;
+
+	err = _init_reset();
+	if (err) {
+		return err;
+	}
+
+	return err;
+}
+
+/** Clean up and reinitialize */
+isoal_status_t isoal_reset(void)
+{
+	isoal_status_t err = ISOAL_STATUS_OK;
+
+	err = _init_reset();
+	if (err) {
+		return err;
+	}
+
+	return ISOAL_STATUS_OK;
```

Comment:
Please align this and isoal_init to avoid needless discrepancies

Reply:
Did, will keep the two functions for now as I see HCI commands calling for both

Reply:
Both functions are too be kept, the comment was just on the needless difference